### DISCUSSION TITLE
Github pages: 200.html support

### DISCUSSION THREAD

**Discussion by pjebs** [community/community#13383] [Other Feature Feedback, Questions, & Ideas]
See: https://github.com/isaacs/github/issues/408

For making single page applications using Github Pages, we need to create a `404.html` file. It works well, except that it accompanies the response with a http status code of 404 (which is generally desirable and as intended).

We propose that a similar `200.html` file is supported that operates in the same manner, except it returns a http status code of 200 instead.

The advantages are substantial when it comes to SEO. My site: https://petersir.com is suffering due to this issue.

The side-effects would need to be mitigated by another configuration option to make it backwards compatible. (But that would have been the case when the 404 file was first implemented). Perhaps also something like a `.nojekyl` file. Perhaps another option in the Github pages dashboard to select which file gets priority if both a `404.html` and a `200.html` are present.

**Comment by jfullstackdev** [comment]
I think this is similar to

https://github.com/orgs/community/discussions/27676

and already answered by the Staff member
